src/test/java/io/jenkins/plugins/analysis/warnings/recorder/TrendCarouselITest.java

```diff
@@ -0,0 +1,271 @@
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Java;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.TrendCarousel;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.TrendCarousel.TrendChartType;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Provides tests for the trend carousel shown on the details page.
+ *
+ * @author Tobias Redl
+ * @author Andreas Neumeier
+ */
+
+public class TrendCarouselITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private HtmlPage webPage;
+
+    /**
+     * Test that severities trend chart is default.
+     */
+    @Test
+    public void shouldShowSeveritiesTrendChartAsDefault() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
+    }
+
+    /**
+     * Test that tools trend chart is next.
+     */
+    @Test
+    public void shouldShowToolsTrendChartAsNext() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        assertThat(carousel.next().equals(carousel.getActive())).isTrue();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.TOOLS);
+    }
+
+    /**
+     * Test that new versus fixed trend chart is previous.
+     */
+    @Test
+    public void shouldShowNewVersusFixedTrendChartAsPrevious() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        assertThat(carousel.previous().equals(carousel.getActive())).isTrue();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+    }
+
+    /**
+     * Test that new versus fixed trend chart is shown after two clicks on next.
+     */
+    @Test
+    public void shouldShowNewVersusFixedTrendChartAfterTwoTimesNext() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        carousel.next(2);
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+    }
+
+    /**
+     * Test that tools trend chart is shown after two clicks on previous.
+     */
+    @Test
+    public void shouldShowToolsTrendChartAfterTwoTimesPrevious() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        carousel.previous(2);
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.TOOLS);
+    }
+
+    /**
+     * Test that tools trend chart is shown after three clicks on next.
+     */
+    @Test
+    public void shouldShowSeveritiesTrendChartAfterThreeTimesNext() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        carousel.next(3);
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
```

Comment:
These tests actually test the page object and not the UI.

Reply:
@uhafner Do you mean that the tests were to small, or that our assertions are not as they should be?

Reply:
No, the comment actually was about the functionality next(2) and next(3). These methods are not in the actual UI, they are a functionality of the page object. 

So navigating step by step and validating each step is the best approach (as you are doing now).